### PR TITLE
Awesome books: plain JavaScript with objects

### PR DESCRIPTION
**Pull request by oliverSCZ**
I did the following:

- Created a GitHub repo entitled, 'awesome-books'.
- Created a new branch, 'add-javascript' from main in accordance with GitHub flow rules.
- Added all files required and followed the necessary instructions (.gitignore, .html, .css, test.md, README.md).
- Committed files (as required).
- Effectively set up a HTML & CSS & JS linter.
- Ensured that irrelevant files were ignored by Git.
- Documented my work professionally and added a descriptive README.md file to the project's repository using the template provided.
- I did that by using JavaScript objects and arrays. And also I did dynamically modifying the DOM and adding basic events.
- Opened this pull request as instructed.

### CODE REVIEW

File: main.js [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+  const title = document.querySelector('.title').value;
+  const author = document.querySelector('.author').value;
+  const book = {
+    title,
+    author,
+  };
+  bookList.push(book);
+  saveDataLocally(bookList);
+  generateBooks();
+};
```
Comment (omar-labana) [inline main.js]: - [x] Please make sure to clear the input field after adding a book so users can add new books.
- [ ] Please make sure to add some validation for empty entries (currently I can submit a book that has no title or author name).

File: main.js [hunk 2]
```diff
@@ -0,0 +1,76 @@
+let bookList = [];
```
Comment (omar-labana) [inline main.js]: - [ ] Kindly add a default collection that has some books when first visiting the page.

**Review by omar-labana — CHANGES_REQUESTED** [review]
Hi @oliverSCZ, 

Excellent work so far! 👌
You have implemented almost all the required features. ✔️ 
There are some cases that you need to adjust but you are almost done! 😀

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._
_N/A_

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by BrianSammit — APPROVED** [review]
# STATUS: APROVED :green_circle: :clap: 

Hello @oliverSCZ, 
Your project is complete!  There is nothing else to say other than... Great job 👏 
Congratulations! 🎉 

Now you can merge this branch and move on :1st_place_medal: 
HAPPY CODING AND GOOD LUCK WITH THE COMMING ONES :crossed_fingers: :four_leaf_clover: 

well-1

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by oliverSCZ** [comment]
Thank you Mr. @omar-labana  and Mr. @BrianSammit  for your feedback and recommendations, much appreciated.

**Pull request merged**